modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/swap/DiscountingFxResetNotionalExchangePricer.java

```diff
@@ -108,19 +111,30 @@ public void explainPresentValue(FxResetNotionalExchange event, RatesProvider pro
   //-------------------------------------------------------------------------
   @Override
   public MultiCurrencyAmount currencyExposure(FxResetNotionalExchange event, RatesProvider provider) {
+    double dfCounterMaturity = provider.discountFactor(event.getCurrency(), event.getPaymentDate());
     LocalDate fixingDate = event.getObservation().getFixingDate();
     FxIndexRates rates = provider.fxIndexRates(event.getObservation().getIndex());
-    double df = provider.discountFactor(event.getCurrency(), event.getPaymentDate());
     if (!fixingDate.isAfter(provider.getValuationDate()) &&
         rates.getFixings().get(fixingDate).isPresent()) {
       double fxRate = rates.rate(event.getObservation(), event.getReferenceCurrency());
-      return MultiCurrencyAmount.of(CurrencyAmount.of(event.getCurrency(), event.getNotional() * df * fxRate));
+      return MultiCurrencyAmount.of(CurrencyAmount.of(event.getCurrency(), event.getNotional() * dfCounterMaturity * fxRate));
     }
+
+    Currency baseCurrency = event.getReferenceCurrency();
+    Currency counterCurrency = event.getCurrency();
+    LocalDate valuationDate = provider.getValuationDate();
+    LocalDate spotDate = FxSwapConvention.of(CurrencyPair.of(baseCurrency, counterCurrency)).calculateSpotDateFromTradeDate(valuationDate,
+        ReferenceData.standard());
```

Comment:
Shouldn't we use valuation reference data here as in other places were we value?